templates/generator_proxy_template/src/contract.rs

```diff
@@ -94,39 +94,35 @@ pub fn execute(
 
 /// ## Description
 /// Receives a message of type [`Cw20ReceiveMsg`] and processes it depending on the received template.
-/// If the template is not found in the received message, then an [`ContractError`] is returned,
+/// If the template is not found in the received message, then a [`ContractError`] is returned,
 /// otherwise it returns a [`Response`] with the specified attributes if the operation was successful
 /// ## Params
 /// * **deps** is an object of type [`DepsMut`].
 ///
-/// * **_env** is an object of type [`Env`].
-///
 /// * **info** is an object of type [`MessageInfo`].
 ///
-/// * **cw20_msg** is an object of type [`Cw20ReceiveMsg`].
+/// * **cw20_msg** is an object of type [`Cw20ReceiveMsg`]. This is the CW20 message to process.
 fn receive_cw20(
     deps: DepsMut,
-    _env: Env,
     info: MessageInfo,
     cw20_msg: Cw20ReceiveMsg,
 ) -> Result<Response, ContractError> {
-    let mut response = Response::new();
+    let response = Response::new();
     let cfg = CONFIG.load(deps.storage)?;
 
     if let Ok(Cw20HookMsg::Deposit {}) = from_binary(&cw20_msg.msg) {
         if cw20_msg.sender != cfg.generator_contract_addr || info.sender != cfg.lp_token_addr {
             return Err(ContractError::Unauthorized {});
         }
-        // deposit tokens in the end reward contract
-        unimplemented!()
+        // TODO: deposit tokens in the end reward contract here
```

Comment:
Better to use macros `todo!("deposit tokens in the end reward contract here")` so the developer will know that something needs to be implemented

Reply:
Done. Thanks